testing/execution.ts

```diff
@@ -1044,7 +1149,8 @@ async function executeSyncFormulaWithContinuations({
       );
     }
     result.push(...response.result);
-    if (chainedCommand) {
+    // Skip Sync here, since we'll handle incremental syncs after this full sync loop happens
+    if (chainedCommand && chainedCommand.type !== FormulaType.Sync) {
```

Comment:
If you type `chainedCommand` at line 1117 as `InterleavedChainedCommandFormulaSpecification` then you can just delete the extra check here since if you have a chained command here it'll be guaranteed to be applicable. This is already the case at runtime, just need to tighten up the types.